templates/terraform/examples/region_network_endpoint_group_appengine.tf.erb

```diff
@@ -0,0 +1,69 @@
+// App Engine Example
+resource "google_compute_global_network_endpoint_group" "<%= ctx[:primary_resource_id] %>" {
+  name                  = "gae-neg"
+  network_endpoint_type = "SERVERLESS"
+  region = "us-central1"
+  app_engine {
+    service = google_app_engine_flexible_app_version.<%= ctx[:primary_resource_id] %>.service
+    version = google_app_engine_flexible_app_version.<%= ctx[:primary_resource_id] %>.version_id
+  }
+}
+
+resource "google_app_engine_flexible_app_version" "<%= ctx[:primary_resource_id] %>" {
+  version_id = "v1"
+  service    = "default"
+  runtime    = "nodejs"
+
+  entrypoint {
+    shell = "node ./app.js"
+  }
+
+  deployment {
+    zip {
+      source_url = google_storage_bucket_object.<%= ctx[:primary_resource_id] %>.self_link
+    }
+  }
+
+  liveness_check {
+    path = "/"
+  }
+
+  readiness_check {
+    path = "/"
+  }
+
+  env_variables = {
+    port = "8080"
+  }
+
+  handlers {
+    url_regex        = ".*\\/my-path\\/*"
+    security_level   = "SECURE_ALWAYS"
+    login            = "LOGIN_REQUIRED"
+    auth_fail_action = "AUTH_FAIL_ACTION_REDIRECT"
+
+    static_files {
+      path = "my-other-path"
+      upload_path_regex = ".*\\/my-path\\/*"
+    }
+  }
+
+  automatic_scaling {
+    cool_down_period = "120s"
+    cpu_utilization {
+      target_utilization = 0.5
+    }
+  }
+
+  noop_on_destroy = true
+}
+
+resource "google_storage_bucket" "<%= ctx[:primary_resource_id] %>" {
+  name = "appengine-static-content"

```

Comment:
We need to be able to run tests in parallel, so you'll want to make this a var like 'neg_name' is (and then add it to the example definition in terraform.yaml as well). Making it a var adds a random value at the end for tests, so we tend to use them for names of resources and other similar fields that need to be unique to a project.

Reply:
fixed